app/controllers/service_types_controller.rb

```diff
@@ -3,8 +3,40 @@ class ServiceTypesController < ApplicationController
 
   # Authentication by default inherited from ApplicationController.
 
+  before_action :authorize_for_districtwide_access_admin, except: [:index]
+
+  def authorize_for_districtwide_access_admin
+    unless current_educator.admin? && current_educator.districtwide_access?
+      error = 'You don\'t have the correct authorization.'
```

Comment:
You can use `raise Exceptions::EducatorNotAuthorized` here and let `application_controller` decide how to translate that to an HTTP response.